b9lab-content/3-main-concepts/02-architecture.md

```diff
@@ -260,35 +292,35 @@ For the sake of the argument, let's assume that we want to tally what happened i
+### What if I like extreme serialization?
 
-As a hacky side-note, this `data` need not be strictly a Merkle root hash. It could well be any bytes as long as the result is deterministic, as mentioned earlier, and collision-resistant, and that the application can recover the state out of it. For instance, if we took our only board and serialized it differently, we could return the board state as such.
+As a hacky side-note, this `data` need not be strictly a Merkle root hash. It could well be any bytes as long as the result is deterministic, as mentioned earlier, and collision-resistant and the application can recover the state out of it. For instance, if you took your only board and serialized it differently, you could return the board state as such.
 
-Namely, we have 64 cells, out of which only 32 are being used. Each cell has either nothing, a black pawn, a black king, a red pawn, or a red king. That's 5 possibilities, which can easily fit in a byte. So we need 32 bytes to describe the board. Since the first bit of a byte is never used when counting to 5, perhaps we can use the very first bit to indicate whose turn it is to play.
+Namely, you have 64 cells, out of which only 32 are being used. Each cell has either nothing, a black pawn, a black king, a red pawn, or a red king. Those are five possibilities, which can easily fit in a byte. So you need 32 bytes to describe the board. Since the first bit of a byte is never used when counting to five, perhaps you can use the very first bit to indicate whose turn it is to play.
 
-So here we go, we have a deterministic blockchain state, collision-resistant since the same value indicates an identical state, no external database to handle, and the full state is always stored in the block header.
+So here you go, you have a deterministic blockchain state, which is collision-resistant since the same value indicates an identical state, no external database to handle, and a full state always stored in the block header.
 
-</ExpansionPanel>
+### Closing remarks
 
-### Closing remarks:
+You have waved your hands as to how you would create a state machine for the checkers game. You surely have already spotted a good number of shortcomings regarding your game blockchain. Let's see:
 
-We have waved our hands as to how we would create a state machine for the checkers game. You surely have already spotted a good number of shortcomings of our game blockchain. Let's see:
+* Anyone, including the opponent, can post an anonymous transaction and play, instead of the intended player. This makes it impossible to know who did what. You need to find a way to **identify the right player**. This will be addressed when we look at [Cosmos SDK's accounts and signatures](./04-accounts.md).
+* Instead of a single game, you would prefer **multiple games** running in parallel, all in a store with an elaborate data model. You need a well-defined store. This will be solved when you look at [Cosmos SDK's key store](linktosection).
```

Comment:
```suggestion
* Instead of a single game, you would prefer **multiple games** running in parallel, all in a store with an elaborate data model. You need a well-defined store. This will be solved when you look at [Cosmos SDK's key store](./10-multistore-keepers).
```
@CitMC